storybookjs/storybook
# React: Support v19 in `react-dom-shim`

## What I did

I updated the React version check in `@storybook/react-dom-shim` to also support React version 19. 

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Update to the latest canary release of React
2. Launch Storybook and open your browser's console
3. Observe 
    ```
    react-16.mjs:3 Uncaught (in promise) TypeError: react_dom__WEBPACK_IMPORTED_MODULE_0__.render is not a function
        at react-16.mjs:3:1
        at new Promise (<anonymous>)
        at renderElement (react-16.mjs:3:1)
        at chunk-HLWAVYOI.mjs:5:1
4. Apply this PR and try again. The error should now go away and your stories should work again


### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
